src/basic/cgroup-util.c

```diff
@@ -425,7 +429,11 @@ int cg_kill(
         if (r == 0)
                 return ret;
 
-        r = cg_kill_items(path, sig, flags, s, log_kill, userdata, "cgroup.threads");
+        /* Opening pidfds for non thread group leaders only works from 6.9 onwards. On older kernels
+         * pidfd_open() fails with EINVAL. Since we might read non thread group leader IDs from
```

Comment:
Well, this would always fail even on 6.9+ since we don't specify `PIDFD_THREAD`. And really, why skip errors that always occur rather than avoiding pidfds completely here?